src/deserialize_sequence.cc

```diff
@@ -339,60 +339,50 @@ Step deserialize_step(const std::filesystem::path& path)
     return step;
 }
 
-std::istream& operator>>(std::istream& stream, Sequence& seq)
+void deserialize_step_setup_script(const std::filesystem::path& path, Sequence& sequence)
 {
-    std::string line;
-    std::string step_setup_script;
-
-    while(std::getline(stream, line, '\n'))
-        step_setup_script += (gul14::trim_right(line) + '\n');
-
-    if (not step_setup_script.empty())
-        seq.set_step_setup_script(step_setup_script);
-
-    return stream;
-}
-
-void deserialize_sequence_impl(const std::filesystem::path& path, Sequence& seq)
-{
-    if (not std::filesystem::exists(path / sequence_lua_filename))
-        return;
+    if (not std::filesystem::exists(path))
+        throw Error(gul14::cat("Path does not exist: '", path.string(), '\''));
 
-    std::ifstream stream(path / sequence_lua_filename);
+    std::string step_setup_script = ""; // set an empty step setup script
 
-    if (not stream.is_open())
-        throw Error(gul14::cat("I/O error: unable to open file '", path.string(), "'"));
+    auto stream = std::ifstream(path / sequence_lua_filename);
+    if (stream.good())
+    {
+        std::string line;
+        while(std::getline(stream, line, '\n'))
+            step_setup_script += (gul14::trim_right(line) + '\n');
```

Comment:
Why is it not just reading the file into a string but removes my dangling whitespace?

Reply:
> Why is it not just reading the file into a string but removes my dangling whitespace?

Ah, that's a good question. I truncate whitespaces on the right side of the string to be more efficient in file savings and to be more reliable in the presented script. I can reduce to 

```diff
- step_setup_script += (gul14::trim_right(line) + '\n');
+ step_setup_script += (line + '\n');
```

Reply:
Hmm. How are the Step's scripts handled?

And regarding line-wise handling...

```c++
    std::ifstream f(path, std::ios::in | std::ios::binary);

    auto const size = std::filesystem::file_size(path);
    std::string step_setup_script(size, '\0');
    f.read(step_setup_script.data(), size);
```

;-) (untested)

Reply:
Thanks for your objection!

What is the benefit of changing the reading function? Is your approach more elegant? Would you expect any not caught exception?

BTW I alter adding new lines with the previous solution:

```c++
step_setup_script += (line + '\n');
```

Reply:
As there is no progress here, I think I'll merge this as-is. In any case this PR is a big step ahead from what we had before.